domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs

```diff
@@ -22,7 +23,8 @@ public ContractServiceImpl(IMeteringPointsClient client, ICertificateIssuingCont
         this.repository = repository;
     }
 
-    public async Task<CreateContractResult> Create(string gsrn, string meteringPointOwner, DateTimeOffset startDate, CancellationToken cancellationToken)
+    public async Task<CreateContractResult> Create(string gsrn, string meteringPointOwner, DateTimeOffset startDate, DateTimeOffset? endDate,
```

Comment:
There is no check here for any existing contracts. There must not be a timely overlap with an existing contract.